modules/core/include/opencv2/core/opencl/opencl_info.hpp

```diff
@@ -64,33 +64,30 @@ static void dumpOpenCLInformation()
 
         std::vector<PlatformInfo> platforms;
         cv::ocl::getPlatfomsInfo(platforms);
-        if (platforms.size() > 0)
-        {
-            DUMP_MESSAGE_STDOUT("OpenCL Platforms: ");
-            for (size_t i = 0; i < platforms.size(); i++)
-            {
-                const PlatformInfo* platform = &platforms[i];
-                DUMP_MESSAGE_STDOUT("    " << platform->name().c_str());
-                Device current_device;
-                for (int j = 0; j < platform->deviceNumber(); j++)
-                {
-                    platform->getDevice(current_device, j);
-                    const char* deviceTypeStr = current_device.type() == Device::TYPE_CPU
-                        ? ("CPU") : (current_device.type() == Device::TYPE_GPU ? current_device.hostUnifiedMemory() ? "iGPU" : "dGPU" : "unknown");
-                    DUMP_MESSAGE_STDOUT( "        " << deviceTypeStr << ": " << current_device.name().c_str() << " (" << current_device.version().c_str() << ")");
-                    DUMP_CONFIG_PROPERTY( cv::format("cv_ocl_platform_%d_device_%d", (int)i, (int)j ),
-                        cv::format("(Platform=%s)(Type=%s)(Name=%s)(Version=%s)",
-                        platform->name().c_str(), deviceTypeStr, current_device.name().c_str(), current_device.version().c_str()) );
-                }
-            }
-        }
-        else
+        if (platforms.empty())
         {
             DUMP_MESSAGE_STDOUT("OpenCL is not available");
             DUMP_CONFIG_PROPERTY("cv_ocl", "not available");
             return;
         }
 
+        DUMP_MESSAGE_STDOUT("OpenCL Platforms: ");
+        for (size_t i = 0; i < platforms.size(); i++)
+        {
+            const PlatformInfo* platform = &platforms[i];
+            DUMP_MESSAGE_STDOUT("    " << platform->name());
+            Device current_device;
+            for (int j = 0; j < platform->deviceNumber(); j++)
+            {
+                platform->getDevice(current_device, j);
+                const char* deviceTypeStr = (current_device.type() == Device::TYPE_CPU) ? "CPU" :
+                    (current_device.type() == Device::TYPE_GPU ? current_device.hostUnifiedMemory() ? "iGPU" : "dGPU" : "unknown");
```

Comment:
These two lines would benefit of even further refactoring, consider extracting a function for getting deviceTypeStr

Reply:
I'd like to avoid touching that part, because device.type() looks odd to me. It's type is int and not cl_device_type. It's also different from Device::TYPE_* enum, because condition of having iGPU is  "device.type() == Device::TYPE_GPU && device.hostUnifiedMemory()" instead of "device.type() == Device::TYPE_IGPU".

Reply:
I just did this refactoring in [#15883](https://github.com/opencv/opencv/pull/15883). I would appreciate a review from you!